src/expressions/await-expr.md

```diff
@@ -0,0 +1,50 @@
+# Await expressions
+
+> **<sup>Syntax</sup>**\
+> _AwaitExpression_ :\
+> &nbsp;&nbsp; [_Expression_] `.` `await`
+
+Await expressions are legal only within `async` contexts, like an
+`async fn` or an `async` block. They operate on a future. Their effect
+is to "pause" the current computation until the given future is ready
+to produce a value.
+
+More specifically, a `f.await` expression has the following effect.
+
+1. evaluate `f` to a future `tmp` using the `IntoFuture:into_future` method;
+2. `tmp` is pinned using `Pin::new_unchecked`;
+3. This pinned future is then polled by invoking the [`Future::poll`] method and
+   passing it the current [task context] (see below);
+3. if the result returns `Poll::Pending`, then the future returns
+   `Poll::Pending`, suspending its state so that, when re-invoked, we will return
```

Comment:
```suggestion
   `Poll::Pending`, suspending its state so that, when re-invoked,
    execution resumes at step 2;
```

Reply:
"re-invoked" to "re-polled"?

Reply:
Sure, that might be better -- I think both work.